libs/dyn/pattern_trie.go

```diff
@@ -0,0 +1,200 @@
+package dyn
+
+import "fmt"
+
+// PatternTrie is a trie data structure for storing and querying patterns.
+// It supports both exact matches and wildcard matches. You can insert [Pattern]s
+// into the trie and then query it to see if a given [Path] matches any of the
+// patterns.
+type PatternTrie struct {
```

Comment:
q - why have this wrapper struct, why not work on *trieNode directly?

Reply:
It's an abstraction. Each tree node is not a valid trie and thus should not have methods like Insert or SearchPath. Only the root node should have them.

Reply:
I was thinking that in order match all patterns in parallel one could walk bundle and patterns struct together. So if you step into "resources" in bundle, you only need to consider up to 2 trieNodes: root.anyKey and root.pathKey["resources"].

This could be useful to implement efficient match for all patterns across a given bundle.

Reply:
Yeah that would be more efficient. If would also require walking the bundle config tree and the trie at the same time which is more complex and the gains are marginal. 

Let's keep it simple for now.. But I can open up the interface.

Reply:
Updated to export trienode